src/module/actor/creature/document.ts

```diff
@@ -428,100 +412,27 @@ abstract class CreaturePF2e<
 
         const systemData = this.system;
         const checkType = systemData.attributes.initiative.ability || "perception";
+        const baseStatistic = checkType === "perception" ? this.perception : this.skills[checkType];
+        const statistic = (() => {
+            if (!baseStatistic) {
+                return new Statistic(this, {
+                    slug: "initiative",
+                    label: "PF2E.Actor.Creature.Spellcasting.InvalidProficiency",
```

Comment:
I think we need a general "failed statistic" translation entry. Let me know where to put it.